packages/contracts-bedrock/scripts/Deploy.s.sol

```diff
@@ -0,0 +1,667 @@
+
+        save("L1CrossDomainMessengerProxy", address(proxy));
+        console.log("L1CrossDomainMessengerProxy deployed at %s", address(proxy));
+
+        address addr = addressManager.getAddress(contractName);
+        if (addr != address(proxy)) {
+            addressManager.setAddress(contractName, address(proxy));
+        }
+
+        require(addressManager.getAddress(contractName) == address(proxy));
+
+        return address(proxy);
+    }
+
+    /// @notice Deploy the OptimismPortalProxy
+    function deployOptimismPortalProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy({
+            _admin: proxyAdmin
+        });
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("OptimismPortalProxy", address(proxy));
+        console.log("OptimismPortalProxy deployed at %s", address(proxy));
+
+        return address(proxy);
+    }
+
+    /// @notice Deploy the OptimismMintableERC20FactoryProxy
+    function deployOptimismMintableERC20FactoryProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy(proxyAdmin);
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("OptimismMintableERC20FactoryProxy", address(proxy));
+        console.log("OptimismMintableERC20FactoryProxy deployed at %s", address(proxy));
+
+        return address(proxy);
+    }
+
+    /// @notice Deploy the L1ERC721BridgeProxy
+    function deployL1ERC721BridgeProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy({
+            _admin: proxyAdmin
+        });
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("L1ERC721BridgeProxy", address(proxy));
+        console.log("L1ERC721BridgeProxy deployed at %s", address(proxy));
+
+        return address(proxy);
+    }
+
+    /// @notice Deploy the SystemConfigProxy
+    function deploySystemConfigProxy() broadcast() public returns (address) {
+        address proxyAdmin = mustGetAddress("ProxyAdmin");
+        Proxy proxy = new Proxy({
+            _admin: proxyAdmin
+        });
+
+        address admin = address(uint160(uint256(vm.load(address(proxy), OWNER_KEY))));
+        require(admin == proxyAdmin);
+
+        save("SystemConfigProxy", address(proxy));
+        console.log("SystemConfigProxy deployed at %s", address(proxy));
+
+        return address(proxy);
+    }
+
+    /// @notice Deploy the DisputeGameFactoryProxy
+    function deployDisputeGameFactoryProxy() broadcast() public returns (address) {
+        if (block.chainid == 900) {
```

Comment:
This is where it'd be nice to feed forge script long args that are passed into context if unrecognized.

eg: `forge script Deploy.s.sol --with-dgf`
and inside the script: `bool set = vm.getArgument("--with-dgf");`

Reply:
This would be interesting but requires a change to foundry. I agree that the devex is better than overloading `run`. Would need a lot of different cheatcodes like `vm.getBoolArg("--with-dgf")` ie for each type